pkg/binding/transport.go

```diff
@@ -11,58 +13,100 @@ import (
 // BindingTransport adapts the transport.Transport interface to binding.Sender and binding.Receiver.
 // This adapter doesn't support the request/response model
 type BindingTransport struct {
-	Sender   Sender
-	Receiver Receiver
+	Sender    Sender
+	Requester Requester
+	Receiver  Receiver
 	// SenderContextDecorators can be used to decorate the context passed to the Sender.Send() method
 	SenderContextDecorators []func(context.Context) context.Context
 	handler                 transport.Receiver
+	converter               transport.Converter
 }
 
 var _ transport.Transport = (*BindingTransport)(nil) // Conforms to the interface
 
-func NewTransportAdapter(s Sender, r Receiver, senderContextDecorators []func(context.Context) context.Context) *BindingTransport {
+func NewRequestingTransport(r Requester, rx Receiver, senderContextDecorators []func(context.Context) context.Context) *BindingTransport {
 	if senderContextDecorators == nil {
 		senderContextDecorators = []func(ctx context.Context) context.Context{}
 	}
-	return &BindingTransport{Sender: s, Receiver: r, SenderContextDecorators: senderContextDecorators}
+	return &BindingTransport{Requester: r, Receiver: rx, SenderContextDecorators: senderContextDecorators}
+}
+
+func NewSendingTransport(s Sender, rx Receiver, senderContextDecorators []func(context.Context) context.Context) *BindingTransport {
+	if senderContextDecorators == nil {
+		senderContextDecorators = []func(ctx context.Context) context.Context{}
+	}
+	return &BindingTransport{Sender: s, Receiver: rx, SenderContextDecorators: senderContextDecorators}
 }
 
 func (t *BindingTransport) Send(ctx context.Context, e event.Event) (context.Context, *event.Event, error) {
 	for _, f := range t.SenderContextDecorators {
 		ctx = f(ctx)
 	}
-	return ctx, nil, t.Sender.Send(ctx, EventMessage(e))
+
+	if t.Sender != nil {
+		return ctx, nil, t.Sender.Send(ctx, EventMessage(e))
+	}
+	// If provided a requester, use it to do request/response.
+	var resp *event.Event
+	msg, err := t.Requester.Request(ctx, EventMessage(e))
+	if err == nil {
+		if rs, _, err := ToEvent(ctx, msg, nil); err != nil {
+			cecontext.LoggerFrom(ctx).Warnw("failed calling ToEvent", zap.Error(err))
+		} else {
+			defer func() {
+				if err := msg.Finish(err); err == nil {
+					cecontext.LoggerFrom(ctx).Warnw("failed calling message.Finish", zap.Error(err))
+				}
+			}()
+			resp = &rs
+		}
+	}
+	return ctx, resp, err
 }
 
-func (t *BindingTransport) SetReceiver(r transport.Receiver) { t.handler = r }
+func (t *BindingTransport) SetReceiver(r transport.Receiver) {
+	t.handler = r
```

Comment:
Look this comment: https://github.com/cloudevents/sdk-go/pull/343/files#r388793129